### ISSUE TITLE
[Improvement] Are We Ready For Version 1.0.0?

### ISSUE THREAD

**Issue by webadict** [demilich1/metastone#75]
Hey guys!

I've got another Pull Request incoming that I think would be a great starting point for the new 1.0.0 version. There's been some great additions to Metastone since I've joined, but I'm really excited about these. However, I figure that I should ask everyone if there's anything missing that should be added now. I've mostly be doing some background cleaning up, as well as fixing several random bugs that were pretty well disguised.

This next PR focuses mainly on removing extra unused Triggers and Spells, and updating those that are there to be much more useful and versatile. Afterwards, I'll try to focus on the documentation, so that custom card makers might be interested in testing out the program. This would give a greater tester base for Metastone, and it might increase its popularity.

Also, thanks for everyone that helps bug test and supports Metastone. I know I'm not the creator of Metastone and that I'm basically a glorified debugger, but I like helping out on this project. I mostly do programming for fun and as a small part of my job, but without everyone here, this wouldn't be nearly as fun. So thanks again.

I want to get the deck editing down, but I am horrendous at XML, and every time I try, I either blow everything up or nothing happens. So, if anyone can get those particular issues out of the Issues list, then we'd really be ready.

Get pumped for 1.0.0!

**Comment by demilich1** [comment]
For me there is one thing I would like to get fixed for 1.0.0 but it is a bit complicated. I opened a separate issue for that: https://github.com/demilich1/metastone/issues/76

**Comment by webadict** [comment]
Anything else left that should be looked at?

**Comment by demilich1** [comment]
We are good to go in my opinion. I will review your changes and merge the pull request in the next few days. Then we are ready for a 1.0.0 release candidate, which will be available on github for testing.

**Comment by webadict** [comment]
I patched Faceless related Aura bugs so they shouldn't be too big of an issue. Unfortunately, there's a separate issue I found involving TurnEndTriggers incorrectly applying. It's a particularly annoying bug for Player 2, so I'll fix it real quick and then do a PR. Also, I'll do one last fix for Faceless while I'm at it.

**Comment by webadict** [comment]
Alright, I've fixed the few issues that I found. That makes the list of changes this:

Cards:
-Iron Sensei is no longer called Iron Sensai... again.
-Astral Communion now grants full mana crystals.
-Poison Seeds no longer summons Treants before processing deathrattles.
-Multiple Hobgoblins apply their buffs now instead of being stingy and only giving one.
-Djinni of Zephys now properly recasts spells on itself even if the conditional parts of the spell are not met.
-Obsidian Destroyer is now a Warrior card instead of a Neutral.
-Misdirection now works on your opponent's minions, instead of waiting for your own to attack your Hero.
-Snake Traps now also works the same.
-...Actually, the list of broken secrets might be too long. Let's just secrets should now work as written.
-Fixed Blood Knight from tripling its Divine Shield buffs.
-Fixed Bolf Ramshield, Animated Armor, and Ice Block to work as Pre-Damage triggers, instead of hard-coded instances.
-Fixed Commanding Shout to work as intended instead of as a global buff for all of a player's units.
-Fixed Alarm-o-bot from occasionally jumping into your hand and replaying itself.
-Fixed Gahz'rilla and Blessed Champion to double temporary Attack temporarily.
-Fixed Lay on Hands to draw 3 cards instead of 1.
-Removed Demon race tag from Majordomo Executus.
-Fixed various summoning battlecries to summon on the correct sides.
-Fixed Dark Wispers being a Classic card instead of a GvG card.
-Fixed Wilfred Fizzlebang to properly set card costs to 0.
-In fact, fixed a lot of card cost modifying cards to work in conjunction with each other. (Aviana, Loatheb, Millhouse Manastorm, Naga Sea Witch, and possibly others.)
-Updated cards texts.

Documentation:
-Added (incomplete) attribute, spell, and trigger documentation.
-Updated other documentation.
-Removed unused documentation.

GUI:
-Improved colors for health and attack values.
-Added multiple damage bubbles for simultaneous damage events.
-Added deck deletion from Deck Builder.
-Added fatigue counter when your deck is empty.
-Added Mega-Windfury identifier.
-Improved the deck list counter.

Internal:
-Added Set Aside Zone for entities that do not belong in the Graveyard.
-Fixed transformations.
-Removed direct entity calls in the environment.
-Improved triggers and events by a lot.
--Triggers now have host targeting types.
--Triggers now have event sources in addition to event targets.
--Triggers now have source players in addition to target players.
-Fixed Game State Value AI thinking too hard about Discover actions.
-Improved Algebraic Operations, Operations, and other enumerators.
-Removed a lot of unnecessary and unused spells, attributes, and triggers.
-Fixed battlecry resolution so that minions are added to the board before applying.

Tests:
-Added Aura tests.
-Added Faceless Manipulator tests for Auras.
